### PR TITLE
Fix View button in leaderboards listing

### PR DESCRIPTION
**Pull request by piotrekpasciak**
Github issue:
https://github.com/AIcrowd/AIcrowd/issues/1215

### CODE REVIEW

File: app/views/leaderboards/_leaderboards.html.erb [hunk 1]
```diff
@@ -53,10 +52,11 @@
...
     <td class="text-right">
-      <% if leaderboard_participant.present? %>
-        <%= link_to challenge_submission_path(challenge, leaderboard.submission_id), class: 'btn btn-secondary btn-sm' do %>
-          View
-        <% end %>
+      <% if leaderboard.submitter.present? && leaderboard.submission.present? %>
+        <%= link_to 'View', challenge_submission_path(challenge, leaderboard.submission_id), class: 'btn btn-secondary btn-sm' %>
+      <% else %>
+        <!-- Submitter or Submission was deleted -->
+        <%= link_to 'View', '#', class: 'btn btn-secondary btn-sm disabled', disabled: true %>
```
Comment (nemaniarjun) [inline app/views/leaderboards/_leaderboards.html.erb]: Please check the [team helpers](https://github.com/AIcrowd/AIcrowd/blob/737f604ac807b86476e9d62f1703114303b2eae8/app/helpers/team_helper.rb) on how we can set tooltip text on the disabled button. 
AFAI remember you just need to set html title attrib

Comment (alkeshrails) [reply app/views/leaderboards/_leaderboards.html.erb]: Agree with @nemaniarjun , we should show title on disabled field.

Comment (piotrekpasciak) [reply app/views/leaderboards/_leaderboards.html.erb]: @nemaniarjun @alkeshrails I've added this comment as HTML 5 title tooltip.

File: app/views/leaderboards/_leaderboards.html.erb [hunk 2]
```diff
@@ -53,10 +52,10 @@
 
     <td class="text-right"><%= local_time(leaderboard.created_at, "%a, %e %b %Y %H:%M") %></td>
     <td class="text-right">
-      <% if leaderboard_participant.present? %>
-        <%= link_to challenge_submission_path(challenge, leaderboard.submission_id), class: 'btn btn-secondary btn-sm' do %>
-          View
-        <% end %>
+      <% if leaderboard.submitter.present? && leaderboard.submission.present? %>
```
Comment (skbly7) [inline app/views/leaderboards/_leaderboards.html.erb]: This will break for UnknownParticipant leaderboards afaik. @nemaniarjun?

Comment (skbly7) [reply app/views/leaderboards/_leaderboards.html.erb]: Tested for UnknownParticipant and this thing will break:

image


```

      <% if participants.empty? %>	      <% if participants.empty? %>
        <%= image_tag default_image_url, class: 'avatar' %>	        <%= image_tag default_image_url, class: 'avatar' %>
        <span class="text-link">Unknown User</span>	        <span class="text-link">Unknown User</span>
      <% elsif participants.length == 1 %>	      <% elsif participants.length == 1 %>
        <%= link_to participant_path(participants[0]), title: participants[0].name, data: {toggle: :tooltip} do %>	        <%= link_to participant_path(participants[0]), title: participants[0].name, data: {toggle: :tooltip} do %>
          <%= image_tag participants[0].image_url, class: 'avatar' %>	          <%= image_tag participants[0].image_url, class: 'avatar' %>
          <span class="text-link"><%= participants[0].name %></span>	          <span class="text-link"><%= participants[0].name %></span>
        <% end %>	        <% end %>
      <% else %>	      <% else %>
```

^ Check line number 14 of this file, might make sense to do like that.

Comment (skbly7) [reply app/views/leaderboards/_leaderboards.html.erb]: Still will have the same error as described above, the if condition thing. Please go again from traceback screenshot above.

Comment (piotrekpasciak) [reply app/views/leaderboards/_leaderboards.html.erb]: @skbly7 Is there a URL address where I can replicate that error? I'm not sure how to replicate.

Comment (piotrekpasciak) [reply app/views/leaderboards/_leaderboards.html.erb]: I think the bug is a result of that some records on production have `submitter_type` of value: OldParticipant and we don't even have class like that.

Comment (piotrekpasciak) [reply app/views/leaderboards/_leaderboards.html.erb]: @skbly7 I've fixed issue described above, by additional if condition:
`leaderboard.submitter_type != 'OldParticipant'`

Fixing this issue properly would require major refactoring, thus I've decided to use fix above.

**Pull request merged**